exercises/error-handling/example.rs

```diff
@@ -0,0 +1,81 @@
+use std::cmp;
+
+#[derive(Eq,PartialEq,Debug)]
+pub enum IceCream {
+    Chocolate,
+    Vanilla,
+    Strawberry,
+}
+
+impl IceCream {
+    /// Convert various string versions of ice cream flavors to a
+    /// proper IceCream enum or return Error
+    pub fn get_flavor(request: &str) -> Result<IceCream, &'static str> {
+        match request {
+            "Chocolate" | "chocolate" | "choc" | "c" | "C" => Ok(Chocolate),
+            "Strawberry" | "strawberry" | "straw" | "s" | "S" => Ok(Strawberry),
+            "Vanilla" | "vanilla" | "v" | "V" | "plain" => Ok(Vanilla),
+            _ => Err("Unknown Flavor"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct IceCreamTruck {
+    // These fields are only pub, because tests are not in the same file
+    // and we need to verify inventory after actions.
+    pub chocolate: u32,
+    pub vanilla: u32,
+    pub strawberry: u32,
+}
+
+use IceCream::*;
+
+impl IceCreamTruck {
+    pub fn new(chocolate: u32, vanilla: u32, strawberry: u32) -> Self {
+        IceCreamTruck { chocolate: chocolate,
+                        vanilla: vanilla,
+                        strawberry: strawberry }
+    }
+
+    /// Buy ice cream and update inventory.
+    /// Return quantity purchased or indication that you could not purchase.
+    pub fn buy(&mut self, qty: u32, flavor: IceCream) -> Option<u32> {
```

Comment:
I know this is an exercise that is supposed to show off options, but I can't help but notice that since we currently never have `Some(0)` (only have `None`) that we might as well just use a straight `u32` here and no option. There needs to be a situation where `Some(0)` could be a valid answer. Perhaps if you try to buy 0 ice creams you will always get a `Some(0)`, that could work. In that case we would just want tests for buying 0 ice creams of a flavour and that will be good!